ceph/ceph_admin/helper.py

```diff
@@ -290,14 +291,61 @@ def generate_rgw_spec(self, spec):
               rgw_frontend_port: 8080
               rgw_realm: east
               rgw_zone: india
+              rgw_frontend_ssl_certificate: create-cert | <contents of crt>
         Returns:
             service_spec
+
+        contents of rgw_spec.yaml file
+
+            service_type: rgw
+            service_id: rgw.india
+            placement:
+              hosts:
+                - node5
+            spec:
+              ssl: true
+              rgw_frontend_ssl_certificate: |
+                -----BEGIN PRIVATE KEY------
+                ...
         """
         template = self._get_template("rgw")
         node_names = spec["placement"].pop("nodes", None)
         if node_names:
             spec["placement"]["hosts"] = self.get_hostnames(node_names)
 
+        # ToDo: This works for only one host. Not sure, how cephadm handles SSL
+        #       certificate for multiple hosts.
+        if spec["spec"].get("rgw_frontend_ssl_certificate") == "create-cert":
```

Comment:
Is any other option can be set to `rgw_frontend_ssl_cert` other than create cert?
If not can we take away comparison part and just validate `spec["spec"].get("rgw_frontend_ssl_certificate")`

Reply:
Another option to pass the certificate string as is which is not processed as we assume it is a proper or invalid.